github/table_github_workflow.go

```diff
@@ -140,3 +150,105 @@ func tableGitHubWorkflowGet(ctx context.Context, d *plugin.QueryData, h *plugin.
 
 	return workflow, nil
 }
+
+func GitHubWorkflowFileContent(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	workflow := h.Item.(*github.Workflow)
+	if workflow.Path == nil {
+		return nil, nil
+	}
+
+	id := d.KeyColumnQuals["id"].GetInt64Value()
+	fullName := d.KeyColumnQuals["repository_full_name"].GetStringValue()
+	owner, repo := parseRepoFullName(fullName)
+	plugin.Logger(ctx).Trace("tableGitHubWorkflowGet", "owner", owner, "repo", repo, "id", id)
+
+	client := connect(ctx, d)
+
+	type GetFileContentResponse struct {
+		content *github.RepositoryContent
+		resp    *github.Response
+	}
+
+	// Get the name of the default branch for the repository
+	workflowUrlParts := strings.Split(*workflow.HTMLURL, "/")
+	defaultBranch := "main"
+	if len(workflowUrlParts) > 6 {
+		defaultBranch = workflowUrlParts[6]
+	}
+
+	// Get workflow file content
+	getFileContent := func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+		content, _, resp, err := client.Repositories.GetContents(ctx, owner, repo, *workflow.Path, &github.RepositoryContentGetOptions{Ref: defaultBranch})
+		return GetFileContentResponse{
+			content: content,
+			resp:    resp,
+		}, err
+	}
+
+	getResponse, err := plugin.RetryHydrate(ctx, d, h, getFileContent, &plugin.RetryConfig{ShouldRetryError: shouldRetryError})
+	if err != nil {
+		// the workflow object exists, but the file is deleted
+		if strings.Contains(err.Error(), "404 Not Found") {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	getResp := getResponse.(GetFileContentResponse)
+	content := getResp.content
+
+	return content, nil
+}
+
+//// TRANSFORM FUNCTIONS
+
+func decodeFileContentBase64(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	repContent, ok := d.Value.(*github.RepositoryContent)
+	if !ok {
+		return nil, nil
+	}
+
+	decodedText, err := base64.StdEncoding.DecodeString(*repContent.Content)
+	if err != nil {
+		plugin.Logger(ctx).Error("github_workflow.decodeFileContentBase64", "Decoding file content error", err)
+		return nil, err
+	}
+
+	return string(decodedText), nil
+}
+
+func toPipeline(buf []byte) (*pipelineModels.Pipeline, error) {
```

Comment:
It would be great if you can add a comment about what this transform function is being used for. Just a short description would be nice.